### PR TITLE
Moved splitAppendedTenantDomain() to an abstract class to avoid duplication.

### PR DESCRIPTION
**Pull request by madumalt**

### CODE REVIEW

File: components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SSOAuthnRequestAbstractValidator.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    protected String splitAppendedTenantDomain(String issuer) throws UserStoreException, IdentityException {
+
+        if (IdentityUtil.isBlank(SAMLSSOUtil.getTenantDomainFromThreadLocal())) {
+            if (issuer.contains("@")) {
+                String tenantDomain = issuer.substring(issuer.lastIndexOf('@') + 1);
+                issuer = issuer.substring(0, issuer.lastIndexOf('@'));
+                if (StringUtils.isNotBlank(tenantDomain) && StringUtils.isNotBlank(issuer)) {
+                    SAMLSSOUtil.setTenantDomainInThreadLocal(tenantDomain);
+                    if (log.isDebugEnabled()) {
+                        log.debug("Tenant Domain: " + tenantDomain + " & Issuer name: " + issuer + "has been " +
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SSOAuthnRequestAbstractValidator.java]: "has been " --> " has been" for log clarity.

Comment (ruwanta) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SSOAuthnRequestAbstractValidator.java]: @mefarazath 
"has been " + "split"
May be we can merge this two.

Comment (mefarazath) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SSOAuthnRequestAbstractValidator.java]: +1.

Comment (madumalt) [reply components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/validators/SSOAuthnRequestAbstractValidator.java]: done

**Pull request merged**